src/app/shared/operators/progressive-list-from-list.ts

```diff
@@ -0,0 +1,44 @@
+import { animationFrames, Observable } from 'rxjs';
+import { map, scan, take } from 'rxjs/operators';
+
+/**
+ * Generate stream providing a list populated by chunks at each interval
+ *
+ * IE: chunkSize = 2, list = [1, 2, 3, 4, 5, 6] streams following values:
+ * 1. [1, 2]             → merge 2 first items
+ * 2. [1, 2, 3, 4]       → merge 2 new items
+ * 3. [1, 2, 3, 4, 5, 6] → merge 2 new items
+ *
+ * IE: chunkSize = 2, startIndex = 3, list = [1, 2, 3, 4, 5, 6] streams following values:
+ * 1. [1, 2, 3]          → until startIndex
+ * 2. [1, 2, 3, 4, 5]    → merge 2 new items
+ * 3. [1, 2, 3, 4, 5, 6] → merge the last remaining item
+ *
+ * @param list array of any value
+ * @param options
+ * @returns Observable of list
```

Comment:
The example makes it quite clear but I don't get what you mean by "merge" in "merge 2 new items"

But I am not sure I understand "Generate stream providing a list populated by chunks at each interval" and I still miss the explanation about its use.

```
Create an observable from an array of values, emitting progressively parts of the array (from its head to the full array).
Used to feed incrementally the UI with data.

Options are: 
- `chunkSize`: the size of the increment (default : 25)
- `startIndex`: the size of the initial increment (if not set, start with `chunkSize`)
- `interval`: observable which emits interval at which the parts are emitted (default to animation frames)
```

BTW, `chunkSize` should maybe called `incrementSize` and `startIndex`, `initialIncrementSize` and probably initialIncrementSize should default to 25.

Reply:
Fixed